grafana-plugin/src/containers/IntegrationForm/IntegrationForm.tsx

```diff
@@ -32,481 +32,613 @@ import { IntegrationHelper } from 'pages/integration/Integration.helper';
-                placeholder="Choose Contact Point"
-                onChange={({ target }) => {
-                  const value = (target as HTMLInputElement).value;
-                  setState({ selectedContactPointOption: value });
-                  setValue('contact_point', value);
-                }}
-              />
-            )}
-          </Field>
-        </div>
-      </VerticalGroup>
-    </div>
-  );
-
-  function onAlertManagerChange(option: SelectableValue<string>) {
-    const contactPointsForCurrentOption = allContactPoints
-      .find((opt) => opt.uid === option.value)
-      .contact_points?.map((cp) => ({ value: cp, label: cp }));
-
-    const newState: Partial<CustomFieldSectionRendererState> = {
-      selectedAlertManagerOption: option.value,
-      contactPoints: contactPointsForCurrentOption,
-    };
-
-    if (isExistingContactPoint) {
-      newState.selectedContactPointOption = null;
-      setValue('contact_point', undefined);
+          </div>
+
+          <div className={styles.selectorsContainer}>
+            <Controller
+              name={FormFieldKeys.AlertManager}
+              control={control}
+              rules={{ required: 'Alert Manager is required' }}
+              render={({ field }) => (
+                <Field
+                  key={'AlertManager'}
+                  invalid={errors[FormFieldKeys.AlertManager] !== undefined}
+                  error={errors[FormFieldKeys.AlertManager]?.message as string}
+                >
+                  <Select
+                    {...field}
+                    options={dataSources}
+                    onChange={onAlertManagerChange}
+                    value={selectedAlertManagerOption}
+                    placeholder="Select Alert Manager"
+                  />
+                </Field>
+              )}
+            />
+
+            <Controller
+              name={FormFieldKeys.ContactPoint}
+              control={control}
+              rules={{ required: 'Contact Point is required', validate: contactPointValidator }}
+              render={({ field }) => (
+                <Field
+                  key={FormFieldKeys.ContactPoint}
+                  invalid={errors[FormFieldKeys.ContactPoint] !== undefined}
+                  error={errors[FormFieldKeys.ContactPoint]?.message as string}
+                >
+                  {isExistingContactPoint ? (
+                    <Select
+                      {...field}
+                      options={contactPoints}
+                      onChange={onContactPointChange}
+                      value={selectedContactPointOption}
+                      placeholder="Select Contact Point"
+                    />
+                  ) : (
+                    <Input
+                      value={selectedContactPointOption}
+                      placeholder="Choose Contact Point"
+                      onChange={({ target }) => {
+                        const value = (target as HTMLInputElement).value;
```

Comment:
shouldn't be:
```
onChange={({ currentTarget }) => {
                        const value = currentTarget.value;
                        ```
                        ?

Reply:
Hmm interesting, good to know. :+1: